src/adapters/local-storage.ts

```diff
@@ -0,0 +1,49 @@
+import { z } from "zod";
+
+import * as constants from "src/constants";
+import { Currency } from "src/domain";
+
+export function getCurrency(): Currency {
+  return getStorageByKey({
+    key: constants.PREFERRED_CURRENCY_KEY,
+    defaultValue: constants.PREFERRED_CURRENCY,
+    parser: z.nativeEnum(Currency),
+  });
+}
+
+export function setCurrency(currency: Currency): void {
+  setStorageByKey({ key: constants.PREFERRED_CURRENCY_KEY, value: currency });
+}
+
+export function getStorageVersion(): number {
```

Comment:
I would remove this storage versioning code as we are going to depend on `localStorage` only for this `preferredCurrency` stuff

Reply:
I was refining some things in the PR and this was one of them.